test/suites/integration/tags_test.go

```diff
@@ -132,23 +166,51 @@ var _ = Describe("Tags", func() {
 			env.ExpectCreated(nodePool, nodeClass, pod)
 			env.EventuallyExpectCreatedNodeCount("==", 1)
 			node := env.EventuallyExpectInitializedNodeCount("==", 1)[0]
-			nodeName := client.ObjectKeyFromObject(node)
+			nodeClaim := env.ExpectNodeClaimCount("==", 1)[0]
 
 			Eventually(func(g Gomega) {
-				node = &corev1.Node{}
-				g.Expect(env.Client.Get(env.Context, nodeName, node)).To(Succeed())
-				g.Expect(node.Annotations).To(HaveKeyWithValue(v1.AnnotationInstanceTagged, "true"))
-			}, time.Minute)
+				g.Expect(env.Client.Get(env.Context, client.ObjectKeyFromObject(nodeClaim), nodeClaim)).To(Succeed())
+				g.Expect(nodeClaim.Annotations).To(HaveKeyWithValue(v1.AnnotationInstanceTagged, "true"))
+			}, time.Minute).Should(Succeed())
 
 			nodeInstance := instance.NewInstance(lo.ToPtr(env.GetInstance(node.Name)))
 			Expect(nodeInstance.Tags).To(HaveKeyWithValue("Name", "custom-name"))
 			Expect(nodeInstance.Tags).To(HaveKey("karpenter.sh/nodeclaim"))
 		})
-	})
-})
+		It("should tag instance with eks:eks-cluster-name tag when the tag doesn't exist", func() {
+			pod := coretest.Pod()
 
-func tagMap(tags []*ec2.Tag) map[string]string {
-	return lo.SliceToMap(tags, func(tag *ec2.Tag) (string, string) {
-		return *tag.Key, *tag.Value
+			env.ExpectCreated(nodePool, nodeClass, pod)
+			env.EventuallyExpectCreatedNodeCount("==", 1)
+			node := env.EventuallyExpectInitializedNodeCount("==", 1)[0]
+			nodeClaim := env.ExpectNodeClaimCount("==", 1)[0]
+
+			Eventually(func(g Gomega) {
+				g.Expect(env.Client.Get(env.Context, client.ObjectKeyFromObject(nodeClaim), nodeClaim)).To(Succeed())
+				g.Expect(nodeClaim.Annotations).To(HaveKeyWithValue(v1.AnnotationInstanceTagged, "true"))
+			}, time.Minute).Should(Succeed())
+
+			_, err := env.EC2API.DeleteTags(&ec2.DeleteTagsInput{
+				Resources: []*string{lo.ToPtr(env.ExpectParsedProviderID(node.Spec.ProviderID))},
+				Tags:      []*ec2.Tag{{Key: lo.ToPtr(v1.EKSClusterNameTagKey)}},
+			})
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("Removing the %s annotation to re-trigger tagging", v1.AnnotationClusterNameTaggedCompatability))
```

Comment:
Shouldn't we first check and expect that it exists so we know we're removing it?

Reply:
Sure, can do